cloud/amazon/ecs_taskdefinition.py

```diff
@@ -120,14 +122,95 @@ def __init__(self, module):
+        existingContainers = list(existing['containerDefinitions']) if 'containerDefinitions' in existing else []
+        expectedContainers = list(expected['containers']) if 'containers' in expected else []
+
+        for containers in [existingContainers, expectedContainers]:
+            containers.sort(key=lambda k: k['name'])
+            for container in containers:
+                # in case some properties are not defined, we put the default values
+                # and sort lists properties for comaprison
+
+                prop_defaults = {
+                    'disableNetworking': False,
+                    'privileged': False,
+                    'readonlyRootFilesystem': False
+                }
+
+                for prop_name, prop_default in prop_defaults.iteritems():
+                    if prop_name not in container:
+                        container[prop_name] = prop_default
+
+                simple_list_props = [
+                    'links',
+                    'command',
+                    'entryPoint',
+                    'dnsServers',
+                    'dnsSearchDomains',
+                    'dockerSecurityOptions']
+
+                for list_prop in simple_list_props:
+                    if list_prop not in container:
+                        container[list_prop] = []
+                    else:
+                        container[list_prop].sort()
+
+                complex_list_props = {
+                    'environment': 'name',
+                    'mountPoints': 'sourceVolume',
+                    'portMappings': 'containerPort',
+                    'volumesFrom': 'sourceContainer',
+                    'ulimits': 'name',
+                    'extraHosts': 'hostname' }
+
+                for list_prop, sort_key in complex_list_props.iteritems():
+                    if list_prop not in container:
+                        container[list_prop] = []
+                    else:
+                        container[list_prop].sort(key=lambda k: k[sort_key])
+
+                complex_list_defaults = {
+                    'mountPoints': {
+                        'readOnly': False
+                    },
+                    'portMappings': {
+                        'protocol': 'tcp'
+                    },
+                    'volumesFrom': {
+                        'readOnly': False
+                    } }
+
+                for list_prop, prop_defaults in complex_list_defaults.iteritems():
+                    for list_elem in container[list_prop]:
+                        for prop_name, prop_default in prop_defaults.iteritems():
+                            if prop_name not in list_elem:
+                                list_elem[prop_name] = prop_default
+
+        # prepare volumes for comaprison
+        existingVolumes = list(existing['volumes']) if 'volumes' in existing else []
+        expectedVolumes = list(expected['volumes']) if 'volumes' in expected else []
```

Comment:
The 'volumes' key may be present in the `expected` dictionary with the value of `NoneType`.

Should line 206 be:
`expectedVolumes = list(expected['volumes']) if 'volumes' in expected and expected['volumes'] is not None else []`